### ISSUE TITLE
Why is the accuracy of classification tasks on dataset(sst2) using this model(state-spaces/mamba-370m-hf) only 0.7?

### ISSUE THREAD

**Issue by xyXsattempt** [state-spaces/mamba#446]
I can reach the accuracy of 0.93 just after changing the base model into google-bert/bert-base-uncased model.

I download the model(state-spaces/mamba-370m-hf) from https://huggingface.co/state-spaces/mamba-370m-hf. 

And here's my code:

import torch
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "1"
from datasets import load_dataset, load_from_disk

dataset = load_from_disk("/data2/home/xuexy/myfiles/local_sst2")
print(dataset["train"][100])

from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("/data2/home/xuexy/memba370")
# tokenizer = AutoTokenizer.from_pretrained("/data2/home/xuexy/myfiles/pretrained_model/bert_uncased")

max_length = 128

def tokenize_function(examples):
    return tokenizer(examples["sentence"], padding="max_length", truncation=True, max_length=max_length)

tokenized_datasets = dataset.map(tokenize_function, batched=True)
# train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(500))
train_dataset = tokenized_datasets["train"]
# vali_dataset = tokenized_datasets["validation"].shuffle(seed=42).select(range(50))
vali_dataset = tokenized_datasets["validation"]
test_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(500))
import torch
from torch import nn

class ClassificationModel(nn.Module):
    def __init__(self, base_model, num_labels=2):
        super(ClassificationModel, self).__init__()
        self.base_model = base_model
        self.classifier = nn.Linear(base_model.config.hidden_size, num_labels)
        self.num_labels = num_labels

    def forward(self, input_ids, attention_mask=None, labels=None):
        outputs = self.base_model(input_ids, attention_mask=attention_mask)
        logits = self.classifier(outputs.last_hidden_state[:, 0, :])  # Use [CLS] token representation
        loss = None
        if labels is not None:
            loss_fct = nn.CrossEntropyLoss()
            labels = labels.type(torch.LongTensor).cuda()
            loss = loss_fct(logits, labels)
        return (loss, logits) if loss is not None else logits

from transformers import AutoModelForSequenceClassification, AutoConfig
from transformers import AutoTokenizer, AutoModel
# Load configuration
config = AutoConfig.from_pretrained("/data2/home/xuexy/memba370", num_labels=2)
# config = AutoConfig.from_pretrained("/data2/home/xuexy/myfiles/pretrained_model/bert_uncased", num_labels=2)
base_model = AutoModel.from_pretrained("/data2/home/xuexy/memba370", config=config)
# base_model = AutoModel.from_pretrained("/data2/home/xuexy/myfiles/pretrained_model/bert_uncased", config=config)
model = ClassificationModel(base_model, num_labels=2)

from transformers import TrainingArguments
import numpy as np
from sklearn.metrics import accuracy_score

import datasets
import evaluate

metric = accuracy_score
def compute_metrics(eval_pred):
    logits, labels = eval_pred
    predictions = np.argmax(logits, axis=-1)
    print("lpgits: ", logits, " labels: ", labels)
    print("eval_pred: ", eval_pred, " predic: ", predictions)
    return {"accuracy": accuracy_score(labels, predictions)}

from transformers import TrainingArguments, Trainer

# Training arguments
training_args = TrainingArguments(
    output_dir="test_trainer",
    evaluation_strategy="steps",
    logging_dir='logs',
    per_device_train_batch_size=64,
    logging_steps=100,
    save_total_limit=2,  # Save only the last 2 checkpoints
    load_best_model_at_end=True,
    num_train_epochs=3,  # Increase the number of epochs if needed
    logging_first_step=True,  # Log the first step
)

trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=train_dataset,
    eval_dataset=vali_dataset,
    compute_metrics=compute_metrics,
)

trainer.train()

model.eval()
with torch.no_grad():
    trainer.evaluate()

**Comment by tridao** [comment]
Bert was trained with masked LM objective. Mamba was trained causal LM objective. The right comparison is between Pythia and Mamba, where both were trained with causal LM objective, but the architectures are different.

**Comment by stargazerteammember** [comment]
Fascinating, I attempted to write the BERT model from scratch, then fine tune it on the sst2 dataset, the the best I could get is 0.73 accuracy on the validation set, and it is driving me crazy
